### PR TITLE
Search for plugins in a few places

### PR DESCRIPTION
**Pull request by prescod**
Snowfakery will now look for a declared plugin or provider in these places:

- the [Python path](https://docs.python.org/3/library/sys.html#sys.path),
- in a `plugins` directory in the same directory as the recipe,
- in a `plugins` directory below the current working directory and
- in a subdirectory of the user's home directory called `.snowfakery/plugins`.

### CODE REVIEW

**Comment by prescod** [comment]
@acrosman . Does this make sense to you?

**Comment by acrosman** [comment]
Yes it does.  Only other thing I might consider would be an command line option to add something for the person who wants some other option.  But this certainly addresses the need I ran into.

**Comment by prescod** [comment]
Yes I'm sure we'll want that at some point but this is probably enough for the next six months or so.

File: snowfakery/plugins.py [hunk 1]
```diff
@@ -102,6 +111,23 @@ def lazy(func: Any) -> Callable:
...
+    cwd_plugins = "./plugins"
+    user_plugins = Path.home() / ".snowfakery/plugins"
+    new_sys_path = [
+        *sys.path,
+        *(str(p) for p in search_paths),
+        str(cwd_plugins),
+        str(user_plugins),
+    ]
+
+    with patch.object(sys, "path", new_sys_path):
```
Comment (davisagli) [inline snowfakery/plugins.py:127]: This looks potentially dangerous if snowfakery is ever embedded in a threaded context, but I'm not sure what to suggest as an alternative.

**Review by davisagli — APPROVED** [review]

**Pull request merged**